src/lib/nnlib.jl

```diff
@@ -10,6 +10,11 @@ end
     return y, Δ -> (Δ * y * (1 - y),)
 end
 
+@adjoint function softplus(x::Real)
+    y = softplus(x)
+    return y, Δ -> (Δ * σ(*),)
```

Comment:
This doesn't look right. Maybe it should be 
```suggestion
    return y, Δ -> (Δ * σ(x),)
```

Reply:
Sorry, a stupid typo. Now fixed.